### ISSUE TITLE
Terminal won’t honor ~/.hushlogin

### ISSUE THREAD

**Issue by exalted** [zed-industries/zed#4827]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Zed will show the “last login” message (e.g., “Last login: Thu Apr  6 08:35:35 on ttys000”) although I have `~/.hushlogin` file in place. (I believe this is a standard Bash behavior?)

### Environment

Zed: v0.80.5 (stable)
OS: macOS 13.3.0
Memory: 64 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by ElijahLynn** [comment]
I just want to add that as of right now, if one adds a `.hushlogin` to the project root, it suppress the `Last login...` AND the `You have mail.` message, in case one ends up here searching for that:

```
Last login: Fri Jan 26 12:20:53 on ttys009
You have mail.
```

**Comment by andrewthauer** [comment]
I'm having an issue where my `$PATH` is evaluated in the wrong order when running in the `zed` terminal. This becomes an issue because `/usr/bin:/bin` get appended (and duplicated) in front of things like `/opt/homebrew/bin` and then bash scripts end up using the very old v3 bash shipped with macOS.

I'm not sure this is related, but based on @mitchellh's detailed explanation, it sounds like it could be related as a root cause. I do not have this problem in the native macOS terminal, iTerm, vscode, or IntelliJ.

**Comment by Thiagoow** [comment]
@Angelk90 Yep, working great for me after: https://github.com/zed-industries/zed/pull/25224
as long as you have a `~/.hushlogin` file or create one!
